src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleHoleESP.kt

```diff
@@ -91,10 +97,138 @@ object ModuleHoleESP : Module("HoleESP", Category.RENDER) {
 
     }
 
+    private object GlowingPlane: Choice("GlowingPlane") {
+        override val parent: ChoiceConfigurable
+            get() = modes
+
+        val outline by boolean("Outline", true)
+
+        val glowHeightSetting by float("GlowHeight", 0.7f, 0f..1f)
+
+
+
+        private val box = Box(0.0, 0.0, 0.0, 1.0, 1.0, 1.0)
+
+        val renderHandler = handler<WorldRenderEvent> { event ->
+            val matrixStack = event.matrixStack
+            val markedBlocks = holes.entries
+
+            renderEnvironmentForWorld(matrixStack) {
+                withDisabledCull {
+                    for ((pos, quality) in markedBlocks) {
+                        val vec3 = Vec3(pos)
```

Comment:
This whole part is nested too deeply which makes the code hard to read. Consider extracting it to a seperate function (i.e. `renderHoleWithGradient`).

Tip: You can let IntelliJ do it for you: Refactor > Extract method